src/jltypes.c

```diff
@@ -3610,21 +3610,23 @@ void jl_init_types(void) JL_GC_DISABLED
     jl_method_instance_type =
         jl_new_datatype(jl_symbol("MethodInstance"), core,
                         jl_any_type, jl_emptysvec,
-                        jl_perm_symsvec(7,
+                        jl_perm_symsvec(8,
                             "def",
                             "specTypes",
                             "sparam_vals",
                             "backedges",
                             "cache",
                             "cache_with_orig",
-                            "precompiled"),
-                        jl_svec(7,
+                            "precompiled",
```

Comment:
I don't think we ever use the upper bits in `precompiled`, so alternatively we could repurpose it into a bitfield. Might merit a name change, though, and perhaps that's not worth it.

Reply:
Made this change, please see if it looks okay.